### PR TITLE
Replace ProgressButton with VaButtonPair in src/platform/forms

### PR DESCRIPTION
**Pull request by powellkerry**
## Summary

- Replace ProgressButton with VaButton and VaButtonPair 
- Update unit tests for the change

## Related issue(s)
- https://app.zenhub.com/workspaces/platform-design-system-5f8de67192551b0012ebb802/issues/gh/department-of-veterans-affairs/vets-design-system-documentation/1533

## Testing done

- Unit tests were updated
- A demo form was set up to test UX and functionality

## Screenshots
Screenshot 2023-03-29 at 8 49 41 AM

Screenshot 2023-03-29 at 8 50 16 AM

## What areas of the site does it impact?
-This change impacts all applications that use /platform/forms/save-in-progress/[ApplicationStatus, FormStartControls, SaveInProgressErrorPage]
-Impacts should not be noticeable to end-user and do not require changes to be made in consuming applications

## Acceptance criteria

- [x ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x ]  No error nor warning in the console.
- [x ]  Events are being sent to the appropriate logging solution
- [x ]  Documentation has been updated (link to documentation)
- [x ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x ]  I added a screenshot of the developed feature
- [x ]  [Accessibility foundational testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### CODE REVIEW

File: src/platform/forms/tests/save-in-progress/01-sip-finish-later.cypress.spec.js [hunk 1]
```diff
@@ -94,9 +100,20 @@ describe('SIP Finish Later', () => {
...
     cy.get('.va-modal').should('be.visible');
-    cy.get('.va-modal .usa-button-primary').click();
+    cy.get('.va-modal va-button-pair')
+      .first()
+      .shadow()
+      .find('va-button')
+      .first()
+      .shadow()
+      .find('button')
+      .click();
```
Comment (jamigibbs) [inline src/platform/forms/tests/save-in-progress/01-sip-finish-later.cypress.spec.js:116]: 🚀 🚀 🚀

**Review by jamigibbs — APPROVED** [review]

**Review by Andrew565 — APPROVED** [review]
LGTM

**Review by Mottie — APPROVED** [review]

**Review by ndsprinkle — APPROVED** [review]

**Review by longmd — APPROVED** [review]
LGTM 🚀

**Review by amponce — APPROVED** [review]
LGTM :shipit:

**Review by CBonade — APPROVED** [review]

**Review by allisonlu — APPROVED** [review]
lgtm!

**Pull request merged**